src/xlate/openroadm/tests/platform.json

```diff
@@ -0,0 +1,70 @@
+  {"synce-reference-clock": {"name": "0", "index": 0, "gearbox": {"name": "1"}}, "input-reference": {"name": "0", "index": 0, "dpll": {"name": "1"}}},
+  {"synce-reference-clock": {"name": "1", "index": 1, "gearbox": {"name": "1"}}, "input-reference": {"name": "1", "index": 1, "dpll": {"name": "1"}}},
+  {"synce-reference-clock": {"name": "2", "index": 2, "gearbox": {"name": "1"}}, "input-reference": {"name": "2", "index": 2, "dpll": {"name": "1"}}},
+  {"synce-reference-clock": {"name": "3", "index": 3, "gearbox": {"name": "1"}}, "input-reference": {"name": "3", "index": 3, "dpll": {"name": "1"}}},
+  {"synce-reference-clock": {"name": "0", "index": 0, "gearbox": {"name": "2"}}, "input-reference": {"name": "4", "index": 4, "dpll": {"name": "1"}}},
+  {"synce-reference-clock": {"name": "1", "index": 1, "gearbox": {"name": "2"}}, "input-reference": {"name": "5", "index": 5, "dpll": {"name": "1"}}},
+  {"synce-reference-clock": {"name": "2", "index": 2, "gearbox": {"name": "2"}}, "input-reference": {"name": "6", "index": 6, "dpll": {"name": "1"}}},
+  {"synce-reference-clock": {"name": "3", "index": 3, "gearbox": {"name": "2"}}, "input-reference": {"name": "7", "index": 7, "dpll": {"name": "1"}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port1", "goldstone": {"module": "piu1", "hostif": 0}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port2", "goldstone": {"module": "piu1", "hostif": 1}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port3", "goldstone": {"module": "piu1", "hostif": 2}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port4", "goldstone": {"module": "piu1", "hostif": 3}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port5", "goldstone": {"module": "piu2", "hostif": 0}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port6", "goldstone": {"module": "piu2", "hostif": 1}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port7", "goldstone": {"module": "piu2", "hostif": 2}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port8", "goldstone": {"module": "piu2", "hostif": 3}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port9", "goldstone": {"module": "piu3", "hostif": 0}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port10", "goldstone": {"module": "piu3", "hostif": 1}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port11", "goldstone": {"module": "piu3", "hostif": 2}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port12", "goldstone": {"module": "piu3", "hostif": 3}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port13", "goldstone": {"module": "piu4", "hostif": 0}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port14", "goldstone": {"module": "piu4", "hostif": 1}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port15", "goldstone": {"module": "piu4", "hostif": 2}}},
+  {"openroadm-port-map": {"openroadm-port-name": "port16", "goldstone": {"module": "piu4", "hostif": 3}}}
```

Comment:
How about adding a field to the existing elements instead of adding new openroadm specific elements?

Something like the following.

```json
{
  "interface": {
    "name": "Interface1/1/1",
    "default": true,
    "pin-mode": "pam4"
  },
  "tai": {
    "module": {
      "name": "piu1",
      "location": "1"
    },
    "hostif": {
      "name": "0",
      "index": 0
    }
  },
  "openroadm": {
    "name": "port1"
  }
}
```

Please note that the number of interfaces may change depending on the configuration (e.g., PAM4 or NRZ).